src/resolve-dependency.ts

```diff
@@ -123,32 +127,65 @@ function getExportsTarget(exports: string | string[] | { [key: string]: string }
   return undefined;
 }
 
-function resolveExportsTarget (pkgPath: string, exp: string | string[] | { [key: string]: string }, subpath: string, job: Job, cjsResolve: boolean): string | undefined {
-  let exports: { [key: string]: string | string[] | { [key: string]: string } };
-  if (typeof exp === 'string' ||
-      typeof exp === 'object' && !Array.isArray(exp) && Object.keys(exp).length && Object.keys(exp)[0][0] !== '.') {
-    exports = { '.' : exp };
+function resolveExportsImports (pkgPath: string, obj: PackageTarget, subpath: string, job: Job, isImports: boolean, cjsResolve: boolean): string | undefined {
+  let matchObj: { [key: string]: PackageTarget };
+  if (isImports) {
+    if (!(typeof obj === 'object' && !Array.isArray(obj) && obj !== null))
+      return undefined;
+    matchObj = obj;
+  } else if (typeof obj === 'string' || Array.isArray(obj) || obj === null ||
+      typeof obj === 'object' && Object.keys(obj).length && Object.keys(obj)[0][0] !== '.') {
+    matchObj = { '.' : obj };
   } else {
-    exports = exp;
+    matchObj = obj;
   }
   
-  if (subpath in exports) {
-    const target = getExportsTarget(exports[subpath], job.exports, cjsResolve);
+  if (subpath in matchObj) {
+    const target = getExportsTarget(matchObj[subpath], job.conditions, cjsResolve);
     if (typeof target === 'string' && target.startsWith('./'))
       return pkgPath + target.slice(1);
   }
-  for (const match of Object.keys(exports)) {
+  for (const match of Object.keys(matchObj).sort((a, b) => a.length > b.length ? -1 : 1)) {
+    if (match.endsWith('*') && subpath.startsWith(match.slice(0, -1))) {
+      const target = getExportsTarget(matchObj[match], job.conditions, cjsResolve);
+      if (typeof target === 'string' && target.startsWith('./'))
+        return pkgPath + target.slice(1).replace(/\*/g, subpath.slice(match.length - 1));
+    }
     if (!match.endsWith('/'))
       continue;
     if (subpath.startsWith(match)) {
-      const target = getExportsTarget(exports[match], job.exports, cjsResolve);
+      const target = getExportsTarget(matchObj[match], job.conditions, cjsResolve);
       if (typeof target === 'string' && target.endsWith('/') && target.startsWith('./'))
-        return pkgPath + match.slice(1) + subpath.slice(match.length);
+        return pkgPath + target.slice(1) + subpath.slice(match.length);
     }
   }
   return undefined;
 }
 
+function packageImportsResolve (name: string, parent: string, job: Job, cjsResolve: boolean): string {
+  if (name !== '#' && !name.startsWith('#/') && job.conditions) {
+    const pjsonBoundary = job.getPjsonBoundary(parent);
+    if (pjsonBoundary) {
+      const pkgCfg = getPkgCfg(pjsonBoundary, job);
+      const { imports: pkgImports } = pkgCfg || {};
+      if (pkgCfg && pkgImports !== null && pkgImports !== undefined) {
```

Comment:
I noticed this line (171) is checking the existence of `pkgCfg` but line 225 is not.

Is that intentional?

Reply:
Yeah it's a redundant check actually. Should be fine to leave.